src/edu/ucsb/nceas/metacat/index/queue/IndexGeneratorTimerTask.java

```diff
@@ -318,125 +240,68 @@ private void index(List<String> metacatIds) {
             boolean followRevisions = false;
-            MetacatSolrIndex.getInstance().submit(pid, sysmeta, isSysmetaChangeOnly, followRevisions);
-            log.info("IndexGenerator.index - submitted the pid " + pid.getValue() +" into RabbitMQ successfully.");
+            MetacatSolrIndex.getInstance()
+                                .submit(pid, sysmeta, isSysmetaChangeOnly, followRevisions);
+            log.info("IndexGenerator.index - submitted the pid " + pid.getValue()
+                       + " into RabbitMQ successfully.");
        }
     }
     
-    /**
-     * Put a delete index task into the index queue
-     * @param id
-     */
-    private void deleteIndex(String id) throws Exception {
-        if(id != null) {
-            SystemMetadata sysmeta = getSystemMetadata(id);
-            Identifier pid = new Identifier();
-            pid.setValue(id);
-            MetacatSolrIndex.getInstance().submitDeleteTask(pid, sysmeta);
-            log.info("IndexGenerator.deleteIndex - submitted the task which deletes pid " + pid.getValue() 
-                     + " into Rabbitmq successfully.");
-        }
-    }
-    
-   
-    
     public void run() {
-    
-        try {
-            log.info("IndexGenerator.run - start to run the index generator timer--------------------------------");
-            if(needReindexFailedEvent) {
-                log.info("IndexGenerator.run - start to reindex previous failed index tasks--------------------------------");
-                reIndexFailedTasks();
-            }
-            if(needReindexSinceLastProcessDate) {
-                log.info("IndexGenerator.run - start to index objects whose modified date is younger than the last process date--------------------------------");
-                Date since = LastReindexDateManager.getInstance().getLastProcessDate();
+        log.info("IndexGenerator.run - start to run the index generator timer");
+        if(needReindexSinceLastProcessDate) {
+            log.info("IndexGenerator.run - start to index objects whose modified date is younger "
+                        + " than the last process date");
+            Date since;
+            try {
+                since = LastReindexDateManager.getInstance().getLastProcessDate();
                 index(since);
+            } catch (IOException | ParseException | InvalidRequest | InvalidToken
+                       | NotAuthorized | NotImplemented | ServiceFailure
+                       | ClassNotFoundException | InstantiationException
+                       | IllegalAccessException | XPathExpressionException
+                       | NotFound | UnsupportedType | SolrServerException
+                       | SAXException | ParserConfigurationException | OREParserException e) {
+                log.error("IndexGenerator.run - Metadata-Index couldn't generate indexes for "
+                           + "those documents which haven't been indexed : " + e.getMessage());
```

Comment:
Adding exceptions to logging is a bit brutal because of the stack traces. We have far too many stack traces for inconsequential stuff as it is. My opinion is that stack traces should only be added to the log at DEBUG levels, except when a true ERROR occurs that would normally be considered a bug in Metacat. Normal operations under INFO debug levels should not produce any stack traces, and any that do show up in the logs should be treated as severe failures to be tracked down and eliminated. In this particular case, I think this is INFO level notification that a particular document wasn't indexed, and not a bug in Metacat. The key is for the log to contain enough info to understand why indexing didn't complete (which probably means differentiating the various exceptions that are rolled together here into those that are transient (and so the doc can be re-queued for indexing) versus permanent (and so need to be added to a dead-letter queue to be dealt with by the repo admins). Even better, this document-level failure to index info probably should go in a more structured data store so it can be delivered to the repo admin/users.

Reply:
Agreed. My comment was because I saw `log.error(`. If it should instead be `log.info(`, then we're fine with no stacktrace